app/mailers/kudo_mailer.rb

```diff
@@ -22,26 +23,30 @@ def batch_kudo_notification(user_id, user_kudos)
         next unless commentable
 
         # If we have a commentable, extract names and process guest kudos text - skip if no kudos givers
-        names = kudo_givers_hash['names']
-        guest_count = kudo_givers_hash['guest_count']
+        names = kudo_givers_hash["names"]
+        guest_count = kudo_givers_hash["guest_count"]
         kudo_givers = []
 
         if !names.nil? && names.size > 0
-          kudo_givers = names
+          # dup so we don't add "a guest" or "5 guests" to the original names
+          # array. If we do that, our kudo_count will be too high whenever we
+          # have both named and guest kudos.
+          kudo_givers = names.dup
           kudo_givers << guest_kudos(guest_count) unless guest_count == 0
-          @kudos_count = names.size + guest_count
+          kudo_count = names.size + guest_count
         else
           kudo_givers << guest_kudos(guest_count).capitalize unless guest_count == 0
```

Comment:
We might want to do https://otwarchive.atlassian.net/browse/AO3-6256 at the same time. Depending on how we fix that the sentence structure may change entirely.

Reply:
Do you want me to add the fix to this pull request? It won't require any changes to the sentence structure with the way I plan to do it.

Reply:
Let's make it a doubleheader. Efficient testing!